Support multiple roles on a contact type (#86)
medic/cht-user-management: README.md

`contact_types.name` | string | The name of the contact_type as it [appears in the app's base_settings.json](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/hierarchy/)
`contact_types.friendly` | string | Friendly name of the contact type
`contact_types.contact_type` | string | The contact_type of the primary contact. [As defined in base_settings.json](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/hierarchy/)

> `contact_types.user_role` | string[] | The [role](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/user-roles/) to assign to the created user. When multiple roles are available, this property acts as a list of allowed roles against which the specified role(s) should be checked.

Reviewer: ```suggestion
`contact_types.user_role` | string[] | A list of allowed [user roles](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/user-roles/). If only one is provided, it will be assigned automatically.
```

`contact_types.replacement_property` | Property | Defines how this `contact_type` is described when being replaced. The `property_name` is always `replacement`. See [ConfigProperty](#ConfigProperty).
`contact_types.place_properties` | Array<ConfigProperty> | Defines the attributes which are collected and set on the user's created place. See [ConfigProperty](#ConfigProperty).
`contact_types.contact_properties` | Array<ConfigProperty> | Defines the attributes which are collected and set on the user's primary contact doc. See [ConfigProperty](#ConfigProperty).

> `contact_types.user_roles_property` | ConfigProperty | Defines the property used for specifying user role(s) when creating users of this contact type. Add flexibility to choose role(s) when there are multiple roles available at this level of your hierarchy. See [ConfigProperty](#ConfigProperty).

Reviewer: I think it is important that we somehow capture the "allowed roles" at each level. Currently, this feature supports free-text input but what if somebody uses the wrong case? What if they use a _CHW role_ for a _supervisor user_ (this can actually result in quite serious bugs)? What if there are trailing spaces on the role, which is super common when working with CSVs?

Ideally, I think the best user experience for this feature would be a set of checkboxes like in the CHT today (see screenshot).  You don't need to implement that UI here in this PR, we can start with free-text; but if we capture the "allowed roles" and validate them, I think that's a good start.

![image](https://github.com/medic/cht-user-management/assets/9014751/e88d042a-abbf-40f3-94ee-16347592d654)

There are probably a few ways to approach this.  But you'll want to validate this propery like the other property types.  You'll probably end up with something like `argument: ["role1", "role2"]`.

Reviewer: [This conversation may be useful](https://github.com/medic/cht-user-management/issues/55#issuecomment-1991383283)

Author: Tried to use it, but since `select_multiple` only accept `parameter` to be object for the moment (not simple array).

Got a simple `role validator` that can do the job, maybe the time to complete `select_multiple` feature.
